Mu2eG4/src/Mu2eG4MTRunManager.cc

```diff
@@ -102,10 +106,9 @@ namespace mu2e {
     if(fSDM)
       { currentRun->SetHCtable(fSDM->GetHCtable()); }
     std::ostringstream oss;
-    //WHAT LIBRARY TO USE?
-    //G4Random::saveFullState(oss);
-    //randomNumberStatusForThisRun = oss.str();
-    //currentRun->SetRandomNumberStatus(randomNumberStatusForThisRun);
+    G4Random::saveFullState(oss);
+    randomNumberStatusForThisRun = oss.str();
+    currentRun->SetRandomNumberStatus(randomNumberStatusForThisRun);
 
```

Comment:
Please fill me in on the plan for storing random number engine state?  The SeedService already has functionality to store the state at the start of each event and I prefer all engines to use it rather than rolling their own.

This brings up another wrinkle - we need to talk more about the management of the engines that are used in the worker threads.  Ideally I would like to manage them with createEngine and the SeedService .  For historical technical reasons, create engine can only be called from the c'tor of a module.  So we would need to make them all in the c'tor, save pointers and pass out the pointers as needed.  I don't know if this causes a big fight with G4 or not.

Reply:
As I mentioned in an email, we cannot call 'createEngine' from a Shared Module, which is what Mu2eG4MT_module.cc is.  There is a method for storing/reading the state of the Worker Run Managers in/from a file.  We can look into implementing this if necessary.

Reply:
Thanks. Let's defer storing and reading state to future work.